java/common/exception/ErrorMessage.java

```diff
@@ -62,17 +62,19 @@
     public static final ErrorMessage INVALID_RULE_WHEN_CONTAINS_DISJUNCTION =
             new ErrorMessage(21, "Rule '%s' 'when' contains a disjunction.");
     public static final ErrorMessage INVALID_RULE_THEN =
-            new ErrorMessage(22, "Rule '%s' 'then' must strictly be either an attribute ownership, or a relation: '%s'.");
+            new ErrorMessage(22, "Rule '%s' 'then' must strictly be a single of either an attribute ownership, or a relation");
+    public static final ErrorMessage INVALID_RULE_THEN_HAS =
+            new ErrorMessage(23, "Rule '%s' 'then' is trying to assign both an attribute type and a variable attribute value: '%s'. Either make the attribute value concrete, or remove the attribute type.");
```

Comment:
awh shiet sorry i forgot to say that you should also shorten this message like in the other PR so it isn't as verbose :) I'm sure you can reword it to fit in the character limit for one line!!

Reply:
shortened!